react/modules/gaEvents.ts

```diff
@@ -45,7 +45,7 @@ export function viewItem(eventData: PixelMessage['data']) {
     items: [item],
   }
 
-  updateEcommerce(eventName, data)
+  updateGA4Ecommerce(eventName, data)
```

Comment:
What do you think of keeping using the existing [`updateEcommerce`](https://github.com/vtex-apps/google-tag-manager/blob/ac112b9e5faa102f2b1b91409c50a15115986a3b/react/modules/updateEcommerce.ts#L15-L27) and modify the calls here to add the `ecommerce` property? e.g.:
```ts
updateEcommerce(eventName, { ecommerce: { data } })
```
Would that achieve the same effect as the proposed `gaUpdateEcommerce`?

I see that `updateEcommerce` does some extra things, such as, [if there's already an `ecommerce: null`](https://github.com/vtex-apps/google-tag-manager/blob/ac112b9e5faa102f2b1b91409c50a15115986a3b/react/modules/updateEcommerce.ts#L16-L18) in the data layer, [it deletes it](https://github.com/vtex-apps/google-tag-manager/blob/ac112b9e5faa102f2b1b91409c50a15115986a3b/react/modules/updateEcommerce.ts#L20-L21) before [pushing a new one](https://github.com/vtex-apps/google-tag-manager/blob/ac112b9e5faa102f2b1b91409c50a15115986a3b/react/modules/updateEcommerce.ts#L24). I don't understand why that is necessary though... Found this was originally implemented [here](https://github.com/vtex-apps/google-tag-manager/pull/84/files). But I wonder if it's worth keeping this same behavior.

Reply:
We can keep using the `updateEcommerce ` but it makes a lot of changes in the dataLayer array, and in [original docs](https://developers.google.com/analytics/devguides/collection/ua/gtm/enhanced-ecommerce?hl=pt-br) has no mention about that manipulation, just a update with `ecommerce: null`, and about `updateEcommerce(eventName, { ecommerce: { data } })
` makes more sense to me give the responsibility for format to the update function and pass just the event data as parameter and makes it more reusable

Reply:
I see. 

After confirming with @icazevedo that most behavior in `updateEcommerce` is not needed, what do you think of us removing all the unnecessary stuff from it? It could have the current implementation of `updateGA4Ecommerce`. Ideally, I was hoping we could have only one of these functions, not two. 

We would then need to update existing uses of `updateEcommerce` in `enhancedEcommerceEvents.ts` to remove the existing `ecommerce` properties.

Reply:
I like this idea. Please go ahead!